framework/src/utils/PetscSupport.C

```diff
@@ -812,6 +812,40 @@ getCommonPetscKeys()
                         true);
 }
 
+bool
+isSNESVI(FEProblemBase & fe_problem)
+{
+#if PETSC_VERSION_LESS_THAN(3, 12, 0)
+
+  PetscOptions & petsc = fe_problem.getPetscOptions();
+
+  int argc;
+  char ** args;
+  PetscGetArgs(&argc, &args);
+
+  std::vector<std::string> cml_arg;
+  for (int i = 0; i < argc; i++)
+    cml_arg.push_back(args[i]);
+
+  if (std::find(petsc.values.begin(), petsc.values.end(), "vinewtonssls") == petsc.values.end() &&
+      std::find(petsc.values.begin(), petsc.values.end(), "vinewtonrsls") == petsc.values.end() &&
+      std::find(cml_arg.begin(), cml_arg.end(), "vinewtonssls") == cml_arg.end() &&
+      std::find(cml_arg.begin(), cml_arg.end(), "vinewtonrsls") == cml_arg.end())
+    return false;
+
+  return true;
+
+#else
+  PetscOptions & petsc = fe_problem.petscOptionsDatabase();
```

Comment:
@fdkong What is the correct way to use petscOptionsDatabase() for PETSc > 3.12? 

Got the following error and realized my usage could be wrong. 
/opt/civet/build_0/moose/framework/src/utils/PetscSupport.C:839:57: error: invalid initialization of reference of type 'Moose::PetscSupport::PetscOptions&' from expression of type 'PetscOptions' {aka '_n_PetscOptions*'}
  839 |   PetscOptions & petsc = fe_problem.petscOptionsDatabase();

Reply:
I think about a little more. I will uniform these interfaces in the future. I realized the old ways should work for PETSc-3.12.x as well.  Just use the old ways for everything at this point. I think it is too much for this PR to handle these dirty things.

Reply:
Changed.